# Загрузка сообщений из Телеграмм-канала 

Скрипт для выгрузки сообщений канала был взят c GitHub: [tg_downloader](https://github.com/bezgubov/tg_downloader)

С его помощью выгружены все сообщения канала [MarketTwits](https://t.me/markettwits) в файл channel_messages.json.

In [1]:
# Выгрузка происходит продолжительное время
#!python tg_downloader.py https://t.me/markettwits --date-from 2026-07-28 --date-to 2026-07-28 --out-dir ./data

In [2]:
import pandas as pd

## Загрузка и первичный обзор данных

In [3]:
df_msgs = pd.read_json('./data/markettwits_posts_2017-01-01_2026-07-27/posts_markettwits_2017-01-01_2026-07-27.json')
df_msgs.info()

<class 'pandas.DataFrame'>
RangeIndex: 370163 entries, 0 to 370162
Data columns (total 8 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   id         370163 non-null  int64  
 1   date_utc   370163 non-null  str    
 2   message    370163 non-null  str    
 3   views      369458 non-null  float64
 4   forwards   369458 non-null  float64
 5   replies    0 non-null       float64
 6   has_media  370163 non-null  bool   
 7   link       370163 non-null  str    
dtypes: bool(1), float64(3), int64(1), str(3)
memory usage: 20.1 MB


In [4]:
df_msgs.head(10)

,id,date_utc,message,views,forwards,replies,has_media,link
0,378988,2026-07-27T15:56:24+00:00,Началось строительство первого в Сочи гостинич...,947.0,1.0,NaN,True,https://t.me/markettwits/378988
1,378987,2026-07-27T15:40:55+00:00,🇷🇺#YDEX #отчетность \nКОНСЕНСУС ИФ: Яндекс во ...,9998.0,29.0,NaN,False,https://t.me/markettwits/378987
2,378986,2026-07-27T15:31:09+00:00,⚠️🇺🇸#акции #сша \nвозобновились распродажи в а...,14558.0,60.0,NaN,True,https://t.me/markettwits/378986
3,378985,2026-07-27T15:18:22+00:00,🇮🇷#иран #геополитика #today \nТрамп после очер...,19400.0,141.0,NaN,True,https://t.me/markettwits/378985
4,378984,2026-07-27T15:09:54+00:00,"⚠️🇮🇷#иран #геополитика \nТрамп заявил, что не ...",21740.0,42.0,NaN,False,https://t.me/markettwits/378984
5,378983,2026-07-27T15:02:22+00:00,🇺🇸#роботы #сша #впк #трамп \nСтартап Foundatio...,22730.0,152.0,NaN,True,https://t.me/markettwits/378983
6,378982,2026-07-27T15:00:35+00:00,⚠️🇺🇸#TSLA \nАкции Tesla сейчас находятся на то...,21527.0,41.0,NaN,True,https://t.me/markettwits/378982
7,378981,2026-07-27T14:52:06+00:00,🇷🇺#геополитика #россия #макро \nШтатная числен...,23247.0,127.0,NaN,False,https://t.me/markettwits/378981
8,378980,2026-07-27T14:50:18+00:00,"🇺🇸#ии #сша \nОжидается, что генеральный директ...",23327.0,15.0,NaN,False,https://t.me/markettwits/378980
9,378979,2026-07-27T14:48:02+00:00,🇷🇺#telegram #россия \n Россия продолжает конта...,23805.0,101.0,NaN,False,https://t.me/markettwits/378979


In [5]:
print(df_msgs['date_utc'].min())
print(df_msgs['date_utc'].max())

2017-11-09T08:38:22+00:00
2026-07-27T15:56:24+00:00


In [6]:
df_msgs.describe(include='all')

,id,date_utc,message,views,forwards,replies,has_media,link
count,370163.000000,370163,370163,3.694580e+05,369458.000000,0.0,370163,370163
unique,NaN,367410,344960,NaN,NaN,NaN,2,370163
top,NaN,2024-12-30T15:37:12+00:00,,NaN,NaN,NaN,False,https://t.me/markettwits/378988
freq,NaN,14,3198,NaN,NaN,NaN,260656,1
mean,189706.027666,NaN,NaN,3.132397e+04,73.716663,NaN,NaN,NaN
std,109411.396693,NaN,NaN,1.858788e+04,152.447575,NaN,NaN,NaN
min,1.000000,NaN,NaN,1.000000e+00,0.000000,NaN,NaN,NaN
25%,94967.500000,NaN,NaN,1.812300e+04,5.000000,NaN,NaN,NaN
50%,189765.000000,NaN,NaN,3.250600e+04,34.000000,NaN,NaN,NaN
75%,284543.500000,NaN,NaN,4.405800e+04,81.000000,NaN,NaN,NaN


Датасет содержит следующие столбцы:
- **id** - идентификатор сообщения в Телеграме (*подлежит удалению*);
- **date_utc** - время публикации сообщения;
- **message** - текст сообщения;
- **views** - количество просмотров (изменяется во времени, актуальное значение можно получить только в моменте, значение можно использовать при построении системы, работающей в режиме реального времени, для определения степени распространения информации и использовать для улучшения работы модели в последующих итерациях разработки, *подлежит удалению*);
- **forwards** - количество ретрансляций сообщения (комментарий тот же, что и для views, *подлежит удалению*);
- **replies** - все значения пустые, для канала не актуально (*подлежит удалению*);
- **has_media** - наличие прикрепленной мультимедийной информации (*подлежит удалению*, мультимедиа на данной итерации не обрабатывается);
- **link** - ссылка на сообщение (*подлежит удалению*).

Выгруженные данные содержат сообщения канала с 8:38 9 ноября 2017 г. по 15:56 27 июля 2026 г., 370163 записей сообщений.
Произведем первичную очистку датасета от отмеченных столбцов, а также от записей без текста сообщений.

In [7]:
drop_columns = ['id', 'views', 'forwards', 'replies', 'has_media', 'link']
df_msgs_clr = df_msgs.drop(columns=drop_columns)
df_msgs_clr.info()

<class 'pandas.DataFrame'>
RangeIndex: 370163 entries, 0 to 370162
Data columns (total 2 columns):
 #   Column    Non-Null Count   Dtype
---  ------    --------------   -----
 0   date_utc  370163 non-null  str  
 1   message   370163 non-null  str  
dtypes: str(2)
memory usage: 5.6 MB


In [8]:
df_msgs_clr = df_msgs_clr[df_msgs_clr['message'] != '']
df_msgs_clr

,date_utc,message
0,2026-07-27T15:56:24+00:00,Началось строительство первого в Сочи гостинич...
1,2026-07-27T15:40:55+00:00,🇷🇺#YDEX #отчетность \nКОНСЕНСУС ИФ: Яндекс во ...
2,2026-07-27T15:31:09+00:00,⚠️🇺🇸#акции #сша \nвозобновились распродажи в а...
3,2026-07-27T15:18:22+00:00,🇮🇷#иран #геополитика #today \nТрамп после очер...
4,2026-07-27T15:09:54+00:00,"⚠️🇮🇷#иран #геополитика \nТрамп заявил, что не ..."
...,...,...
370154,2017-11-09T13:36:01+00:00,Индекс ММВБ может завершить в плюсе четвертую ...
370155,2017-11-09T13:35:05+00:00,Санация банка «Открытие» и Бинбанка не создала...
370156,2017-11-09T13:01:01+00:00,Gold demand slides to eight-year low in third ...
370157,2017-11-09T12:55:55+00:00,«Россети» выступают за переход на долгосрочный...


Предварительно очищенный датасет сообщений канала содержит 366965 сообщений. Для дальнейшей обработки и анализа данных выгружаем датасет в файл.

In [9]:
df_msgs_clr.to_csv('./data/tg_messages.csv', index=False)